subsys/caf/modules/sensor_sampler.c

```diff
@@ -122,20 +249,46 @@ static int sample_sensors(int64_t *next_timeout)
 			}
 		}
 
-		if (*next_timeout > sd->sample_timeout) {
-			*next_timeout = sd->sample_timeout;
+		if (atomic_get(&sd->state) == SENSOR_ACTIVE) {
+			if (*next_timeout > sd->sample_timeout) {
+				*next_timeout = sd->sample_timeout;
+			}
 		}
 	}
 
 	return err;
 }
 
+static int sensor_trigger_init(const struct sensor_config *sc, struct sensor_data *sd)
+{
+	if (IS_ENABLED(CONFIG_ASSERT)) {
+		unsigned int timeout = sc->trigger->activation.timeout_ms;
```

Comment:
We may consider validating other values too (since they are all defined by user configuration file)

Reply:
I was thinking about checking if sampling period is multiplication of trigger timeout but in the end i removed that.